README.md

```diff
@@ -310,73 +333,92 @@ The example requires two trading accounts: `buyer` and `seller`:
+        --node "$NODE" \
         --home "$PIO_NODE" \
         --chain-id "$CHAIN_ID" \
         --keyring-backend test \
         --gas auto \
         --gas-prices 1905nhash \
-        --gas-adjustment 2 \
+        --gas-adjustment 1.2 \
         --testnet \
-        --yes)
-    $ export CODE_ID=$(jq '.logs[0].events[] | select(.type == "store_code").attributes[] | select(.key == "code_id").value | tonumber' <<< "$store_result")
+        --yes \
+        --output json | jq
     ```
+   Find code_id in the result and set
+   ```bash
+   $ export CODE_ID=<<code_id>>
+   ```
 
 3. Instantiate the contract, binding the name `ats-ex.sc.pb` to the contract address:
 
+   _NOTE: It is assumed that the parent name `sc.pb` already exists and is unrestricted._
+
+   <details>
+     <summary>If the name `sc.pb` does not exist, it can be created like so:</summary>
+   
+      ```bash
+      $ provenanced tx name bind "sc" "$NODE0" "pb" \
+          --unrestrict \
+          --from "$NODE0" \
+          --node "$NODE" \
+          --keyring-backend test \
+          --home "$PIO_NODE" \
+          --chain-id "$CHAIN_ID" \
+          --gas-prices 1905nhash \
+          --gas-adjustment 2 \
+          --testnet \
+          --yes
+      ```
+   </details>
+
     ```bash
     $ provenanced tx wasm instantiate "$CODE_ID" \
        '{"name":"ats-ex", "bind_name":"ats-ex.sc.pb", "base_denom":"gme.local", "convertible_base_denoms":[], "supported_quote_denoms":["usd.local"], "approvers":[], "executors":["'$NODE0'"], "ask_required_attributes":[], "bid_required_attributes":[], "price_precision": "0", "size_increment": "1"}' \
         --admin "$NODE0" \
         --label "ats-ex" \
         --from "$NODE0" \
+        --node "$NODE" \
         --home "$PIO_NODE" \
         --chain-id "$CHAIN_ID" \
         --keyring-backend test \
         --gas auto \
         --gas-prices 1905nhash \
-        --gas-adjustment 2 \
+        --gas-adjustment 1.2 \
         --testnet \
         --yes
     ```
 
-   _NOTE: It is assumed that the parent name `sc.pb` already exists and is unrestricted._
-
-   <details>
-     <summary>If the name `sc.pb` does not exist, it can be created like so:</summary>
+4. Get the address of the instantiated contract.
 
-   ```bash
-   $ provenanced tx name bind "sc" "$NODE0" "pb" \
-       --unrestrict \
-       --from "$NODE0" \
-       --keyring-backend test \
-       --home "$PIO_NODE" \
-       --chain-id "$CHAIN_ID" \
-       --gas-prices 1905nhash \
-       --gas-adjustment 2 \
-       --testnet \
-       --yes
-   ```
-   </details>
+   _NOTE: the result when you instantiate sc in step 3 contain the contract_address. OR, `provenanced query wasm list-contracts-by-code $CODE_ID` and get the last one._
 
-4. Get the address of the instantiated contract:
+   Then do:
 
    ```bash
-   $ CONTRACT_ADDRESS=$(provenanced q name resolve "ats-ex.sc.pb" --chain-id "$CHAIN_ID" --testnet | awk '{print $2}')
+   $ export CONTRACT_ADDRESS=<<contract_address>>
```

Comment:
If the contract address is in the output, I'd prefer to use `jq` to pick this out as well. I say that, because I used `jq` in the ats setup script in https://github.com/FigureTechnologies/exchange-ops/blob/main/bin/setup-ats-smart-contract.sh to automate things like creating names, storing and instantiating the contract, etc. and using something like `jq` allows us to automate these steps.

Reply:
yeah, i struggled, since first couple time the query not working, but because the way we do here, i did not see the error, instead it just set the code_id to "to", because the whole sentence is "Failed to", that is why i decide remove this so whoever using this can see the output and understand what it created, yeah, i am working on the exchange-ops part, and yeah, have to abstract the address in order to run the following queries

Reply:
@ysong-figure sounds good. As Bobak said here https://github.com/FigureTechnologies/ats-smart-contract/pull/115#discussion_r1247269698 , you can pull the contract address from the output of `tx instantiate`. If you're having trouble updating the `jq` query / don't have time, if you capture the JSON output of `tx instantiate` and share it with me, I'll figure out the updated the query